apps/studio/src/server/modules/collection/collection.router.ts

```diff
@@ -62,28 +63,70 @@ export const collectionRouter = router({
           resourceId: !!parentFolderId ? String(parentFolderId) : null,
         })
 
-        const result = await db
-          .insertInto("Resource")
-          .values({
-            permalink,
-            siteId,
-            type: ResourceType.Collection,
-            title: collectionTitle,
-            parentId: parentFolderId ? String(parentFolderId) : null,
-            state: ResourceState.Published,
-          })
-          .returning(defaultCollectionSelect)
-          .executeTakeFirstOrThrow()
-          .catch((err) => {
-            if (get(err, "code") === PG_ERROR_CODES.uniqueViolation) {
+        const result = await db.transaction().execute(async (tx) => {
+          const user = await db
```

Comment:
issue: I think should use `tx` and not `db`? Same for the parentFolder one.

Reply:
no real concerns, adrian has a point about shifting it out of the `tx` cos this shouldn't be all or nothing; i think that one more concerning so will shift it out